dask_optuna/storage.py

```diff
@@ -253,53 +269,71 @@ def read_trials_from_remote_storage(
             :exc:`KeyError`:
                 If no study with the matching ``study_id`` exists.
         """
-        return self.get_storage(storage).read_trials_from_remote_storage(
+        return self.get_storage(storage_name).read_trials_from_remote_storage(
             study_id=study_id
         )
 
 
-def register_with_scheduler(dask_scheduler):
+def register_with_scheduler(dask_scheduler=None, storage=None, name=None):
     if "optuna" not in dask_scheduler.extensions:
-        OptunaSchedulerExtension(dask_scheduler)
+        ext = OptunaSchedulerExtension(dask_scheduler)
+    else:
+        ext = dask_scheduler.extensions["optuna"]
+
+    if name not in ext.storages:
+        ext.storages[name] = optuna.storages.get_storage(storage)
 
 
 class DaskStorage(optuna.storages.BaseStorage):
     """ Implements Optuna Storage API """
 
-    def __init__(self, client=None, storage=None):
+    def __init__(self, storage=None, name=None, client=None):
+        self.storage = storage
+        self.name = name or f"dask-storage-{uuid.uuid4().hex}"
```

Comment:
I'm curious, why not reuse storages?  This would make it easy to continue building on existing studies.

Studies are already named, and that seems like the split that one should use to separate things if necessary (or at least it seems that way to me anyway).

My guess is that it is common to pick back up on a pre-existing study, or query about results of an existing study if one exists.  My guess is that having `DaskStorage()` point to the same in-memory storage is more likely going to be the preferred behavior than otherwise.

Reply:
This is primarily to make `DaskStorage` have a similar behavior as other Optuna storage classes when not being reused in a study. For example:

```python
storage_1 = InMemoryStorage()
study_1 = optuna.create_study(..., storage=storage_1, ...)
storage_2 = InMemoryStorage()
study_2 = optuna.create_study(..., storage=storage_2, ...)
```

will create two different `InMemoryStorage` instances that are each used in their own study. But creating multiple `DaskStorage` instances today will reuse the same underlying `InMemoryStorage` instance

```python
storage_1 = DaskStorage()   # Creates a new InMemoryStorage instance
study_1 = optuna.create_study(..., storage=storage_1, ...)
storage_2 = DaskStorage()   # Reuses the same InMemoryStorage instance as storage_1
study_2 = optuna.create_study(..., storage=storage_2, ...)
```

which may be unexpected to users. In particular, when using [`pickle` or `joblib` to save a study](https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-save-and-resume-studies) to disk

Reply:
If users reuse a `DaskStorage` instance across multiple studies by explicitly passing the instance to `study.create_study`, then things already behave similarly to other storage classes today